core/dbt/include/global_project/macros/adapters/timestamps.sql

```diff
@@ -0,0 +1,73 @@
+{%- macro current_timestamp() -%}
+    {{ adapter.dispatch('current_timestamp', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp() %}
+    current_timestamp
+{% endmacro %}
+
+{%- macro current_timestamp_in_utc() -%}
+    {{ adapter.dispatch('current_timestamp_in_utc', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__current_timestamp_in_utc() %}
+    {{ convert_timezone(target_tz="UTC",
+        timestamp=current_timestamp())}}
+{% endmacro %}
+
+{%- macro snapshot_get_time() -%}
+    {{ adapter.dispatch('snapshot_get_time', 'dbt')() }}
+{%- endmacro -%}
+
+{% macro default__snapshot_get_time() %}
+    {{ current_timestamp() }}
+{% endmacro %}
+
+{%- macro convert_timezone(source_tz, target_tz, timestamp) -%}
+    {%- if not target_tz is string -%}
+        {{ exceptions.raise_compiler_error("'target_tz' must be a string") }}
+    {%- else -%}
+        {{ adapter.dispatch('convert_timezone', 'dbt') (source_tz, target_tz, timestamp) }}
+    {%- endif -%}
+
+{%- endmacro -%}
+
+{%- macro default__convert_timezone(source_tz, target_tz, timestamp) -%}
+    {%- if not source_tz -%}
+        {{ timestamp }} at time zone '{{ target_tz }}'
+    {%- else -%}
+        {{ timestamp }} at time zone '{{ source_tz }}' at time zone '{{ target_tz }}'
+    {%- endif -%}
+{%- endmacro -%}
+
+
+---------------------------------------------
+
+/* {#
+    DEPRECATED: DO NOT USE IN NEW PROJECTS
+
+    This is ONLY to handle the fact that Snowflake + Postgres had functionally
+    different implementations of {{ dbt.current_timestamp }} + {{ dbt_utils.current_timestamp }}
+
+    If you had a project or package that called {{ dbt_utils.current_timestamp() }}, you should
+    continue to use this macro to guarantee identical behavior on those two databases.
+#} */
+
+{% macro current_timestamp_backcompat() %}
+    {{ return(adapter.dispatch('current_timestamp_backcompat', 'dbt')()) }}
+{% endmacro %}
+
+{% macro default__current_timestamp_backcompat() %}
+    {{ return(adapter.dispatch('current_timestamp', 'dbt')()) }}
+{% endmacro %}
+
+-- TODO move to dbt-postgres
+{% macro postgres__current_timestamp_backcompat() %}
+    current_timestamp::{{ type_timestamp() }}
+    
+{% endmacro %}
+
+-- TODO move to dbt-snowflake
```

Comment:
move to `dbt-snowflake` (new PR in other repo) before merge

Reply:
if we do it before merge we can't add the functional testing defined in this merge: https://github.com/dbt-labs/dbt-snowflake/pull/273

Reply:
looks like the macro is already over there! that's the only place the `snowflake__` version needs to live. (or maybe I'm not following what you mean)

Reply:
just noticed it's the same as what's in there! Also I think the order is:
1. Merge this
2. Add adapters timestamp macros
3. remove timestamp code from dbt-utils
I *think* we can do steps 2 & 3 concurrently

Reply:
Close. Here's the workflow (start to finish) that we typically use for changes across core + adapters:
1. Open PR for dbt-core (like you did [here](https://github.com/dbt-labs/dbt-core/pull/5838) already)
1. Open PRs for each of the adapters (like you did [here](https://github.com/dbt-labs/dbt-snowflake/pull/273/files), etc already)
1. _Temporarily_ update the `dev-requirements.txt` file within each adapter to point to your PR in Step 1, like [this](https://github.com/dbt-labs/dbt-bigquery/pull/308/commits/7f795796abe4e15b8627d6cf086accce772b5685).
1. Once all PRs are passing CI, reviewed, and approved, then merge the dbt-core PR
1. Undo the temporary update to `dev-requirements.txt` in each of the adapter PRs (so that they point at the `main` branch of dbt-core again) like [this](https://github.com/dbt-labs/dbt-bigquery/pull/313/commits/27d28176b0148263cb05b41be2230ffc43f42700)
1. Once CI passes, then all the adapter PRs can be merged as well.

Reply:
Here's the full life-cycle of a similar adapter PR for reference:
https://github.com/dbt-labs/dbt-bigquery/pull/313/commits

And the associated dbt-core PR:
https://github.com/dbt-labs/dbt-core/pull/5875/commits

Reply:
It would probably be handy if we write out this workflow in some centralized place for easier reference.

Reply:
that's a cool process, however I'm not sure we need to do that here since we haven't removed these macros from dbt-utils yet

Reply:
We've typically coordinated the changes in dbt-core and each dbt adapter in concert (as described above).

The alternative would be to merge a dbt-core in isolation and then do each adapter PR separately afterwards.

⚠️ A situation that can arise when merging an isolated feature to dbt-core:
- the implementation might need to change based on what we learn  and experience within the adapter PRs

The advantage of coordinating the adapter+core PRs is preventing those types of situations.

### In the days of yore
Historically, our adapter code was in the same repo as core (in the [plugins](https://github.com/dbt-labs/dbt-core/tree/main/plugins) directory). In those days (not yet even one year ago!), you'd just do a single PR 'cause everything was in the same repo.

The adapters moved out to their own repos in https://github.com/dbt-labs/dbt-core/pull/3969:
<img width="452" alt="image" src="https://user-images.githubusercontent.com/44704949/191970143-4749b2c2-3eb2-479a-adf7-e2ee523368b7.png">

(There's an Easter egg in #3969 for Billy Joel fans (like me!))